pkg/reconciler/route/resources/ingress.go

```diff
@@ -169,17 +171,31 @@ func getChallengeHosts(challenges []netv1alpha1.HTTP01Challenge) map[string]netv
 	return c
 }
 
-func routeDomain(ctx context.Context, targetName string, r *servingv1.Route, visibility netv1alpha1.IngressVisibility) (string, error) {
+func routeDomain(ctx context.Context, targetName string, r *servingv1.Route, visibility netv1alpha1.IngressVisibility) ([]string, error) {
 	hostname, err := domains.HostnameFromTemplate(ctx, r.Name, targetName)
 	if err != nil {
-		return "", err
+		return []string{}, err
 	}
 
 	meta := r.ObjectMeta.DeepCopy()
 	isClusterLocal := visibility == netv1alpha1.IngressVisibilityClusterLocal
 	labels.SetVisibility(meta, isClusterLocal)
 
-	return domains.DomainNameFromTemplate(ctx, *meta, hostname)
+	domain, err := domains.DomainNameFromTemplate(ctx, *meta, hostname)
+	if err != nil {
+		return []string{}, err
+	}
+	domains := []string{domain}
+	if isClusterLocal {
+		// cluster local domains to support short domains.
+		// e.g. hello.default.svc.cluster.local supports hello.default.svc and hello.default.
+		localDomainSuffix := ".svc." + networkpkg.GetClusterDomainName()
+		if strings.HasSuffix(domain, localDomainSuffix) {
+			domains = append(domains, strings.TrimSuffix(domain, "."+networkpkg.GetClusterDomainName())) // hello.default.svc
+			domains = append(domains, strings.TrimSuffix(domain, localDomainSuffix))                     // hello.default
```

Comment:
This copies the same bug we have currently around numerical TLDs.

If the namespace is a number (allowed) then we should drop this and only add `foo.1234.svc` IIUC?

Reply:
@mattmoor 
Yes, we should do that.
And I realized that we could use [`ExpandedHosts()`](https://github.com/knative/networking/blob/a313c6db9c6d4ad347f5b829d2534836bd80f9d0/pkg/ingress/ingress.go#L83-L100) in networking pkg for here. So I will implement it in `ExpandedHosts()`. (That is what you suggested in https://github.com/knative/serving/issues/7742#issuecomment-675097437 actually.)

Reply:
On this context, I've sent a PR for the change in ExpandedHosts() - https://github.com/knative/networking/pull/164 
Should we change the logic in this code too?